plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -285,60 +507,341 @@ public boolean deleteNuageVspDevice(DeleteNuageVspDeviceCmd cmd) {
+            }
+        }
+
+        if (validateDomains) {
+            auditDomainsOnVsp(host, true, false);
         }
     }
 
-    public class NuageVspSyncTask implements Runnable {
+    private boolean auditDomainsOnVsp(HostVO host, boolean add, boolean remove) {
+        List<NuageVspDeviceVO> nuageVspDevices = _nuageVspDao.listByHost(host.getId());
+        if (!CollectionUtils.isEmpty(nuageVspDevices)) {
+            _hostDao.loadDetails(host);
+            List<DomainVO> allDomains = _domainDao.listAll();
+            for (DomainVO domain : allDomains) {
+                try {
+                    SyncDomainCommand cmd = new SyncDomainCommand(domain.getUuid(), domain.getName(), domain.getPath(), add, remove);
+                    SyncDomainAnswer answer = (SyncDomainAnswer) _agentMgr.easySend(host.getId(), cmd);
+                    return answer.getSuccess();
+                } catch (Exception e) {
+                    s_logger.warn(e.getMessage());
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
 
-        private String nuageVspEntity;
+    private String findNuageVspCmsIdForDevice(long deviceId, ConfigurationVO cmsIdConfig) {
+        String configValue = cmsIdConfig.getValue();
+        if (StringUtils.isNotBlank(configValue)) {
+            String[] configuredNuageVspDevices = configValue.split(";");
+            for (String configuredNuageVspDevice : configuredNuageVspDevices) {
+                if (configuredNuageVspDevice.startsWith(deviceId + ":")) {
+                    return configuredNuageVspDevice.split(":")[1];
+                }
+            }
+        }
+        return null;
+    }
 
-        public NuageVspSyncTask(String nuageVspEntity) {
-            this.nuageVspEntity = nuageVspEntity;
+    public List<String> getDnsDetails(Network network) {
+        List<String> dnsServers = null;
+        Boolean configureDns = Boolean.valueOf(_configDao.getValue(NuageVspManager.NuageVspConfigDns.key()));
+        if (configureDns) {
+            Boolean configureExternalDns = Boolean.valueOf(_configDao.getValue(NuageVspManager.NuageVspDnsExternal.key()));
+            DataCenterVO dc = _dataCenterDao.findById(network.getDataCenterId());
+            dnsServers = new ArrayList<String>();
+            if (configureExternalDns) {
+                if (dc.getDns1() != null && dc.getDns1().length() > 0) {
+                    dnsServers.add(dc.getDns1());
+                }
+                if (dc.getDns2() != null && dc.getDns2().length() > 0) {
+                    dnsServers.add(dc.getDns2());
+                }
+            } else {
+                if (dc.getInternalDns1() != null && dc.getInternalDns1().length() > 0) {
+                    dnsServers.add(dc.getInternalDns1());
+                }
+                if (dc.getInternalDns2() != null && dc.getInternalDns2().length() > 0) {
+                    dnsServers.add(dc.getInternalDns2());
+                }
```

Comment:
Use ``com.google.common.base.Strings.isNullOrBlank`` to check if the strings are blank or null in all of the ``if`` conditions above.